src/Microsoft.Security.Utilities.Core/IdentifiableScan.cs

```diff
@@ -65,27 +65,64 @@ public class IdentifiableScan : ISecretMasker, IDisposable
         byte[] output,
         ref long outputLength);
 
-    private IntPtr scan;
-    private readonly Dictionary<string, ISet<string>> idToSignaturesMap;
     private readonly Dictionary<string, IList<RegexPattern>> signatureToPatternsMap;
+    private readonly Dictionary<string, ISet<string>> idToSignaturesMap;
+    private SecretMasker backupSecretMasker;
     private List<string> orderedIds;
-    
+    private IntPtr scan;
+
+    private static readonly ISet<string> HighPerformanceEnabledSignatures = new HashSet<string>(new string[]
+        {
+            "+ASt",
+            "+ASt",
+            "AzCa",
+            "7Q~",
+            "8Q~",
+            "AzFu",
+            "ACDb",
+            "+ABa",
+            "AzSe",
+            "AzSe",
+            "+AMC",
+            "+ASb",
+            "+AEh",
+            "+ARm",
+            "+ACR",
+            "+AIoT",
+            "APIM",
+            "AZEG",
+            "JQQJ",
+        });
+
     public IdentifiableScan(IEnumerable<RegexPattern> regexPatterns, bool generateCorrelatingIds)
     {   
-        this.generateCorrelatingIds = generateCorrelatingIds;
-        this.idToSignaturesMap = new Dictionary<string, ISet<string>>();
         this.signatureToPatternsMap = new Dictionary<string, IList<RegexPattern>>();
+        this.idToSignaturesMap = new Dictionary<string, ISet<string>>();
+        this.generateCorrelatingIds = generateCorrelatingIds;
         this.orderedIds = new List<string>();
 
         foreach (RegexPattern pattern in regexPatterns)
         {
             if (pattern.Signatures == null)
             {
+                this.backupSecretMasker ??= new SecretMasker(regexSecrets: null,
```

Comment:
Any regex with no signature is not a candidate for the high-perf detection, so we simply slam it into the back-up engine.